### ISSUE TITLE
when broadcast terminator duplicates the keys in others terminals

### ISSUE THREAD

**Issue by jhhenriquee** [gnome-terminator/terminator#504]
I've recently installed Pop!_OS 21.04 and after installed terminator, I end up with this problem.
When in broadcast, all characters that I type, duplicates in the others terminals.

The version of terminator is v 2.1.0. Was installed directly in the pop shop.
I don't know how to install another version.
I've tried to remove and intall with apt. Same thing.

I clicked in preferences to see one option that I could try. Clicked in about and end up here.

![image](https://user-images.githubusercontent.com/71674873/132110736-03648718-851d-49b3-883b-16f404c4f22c.png)

**Comment by mattrose** [comment]
I'm not familiar with what is in the PopOS repositories, would you mind cloning the repository to your and running `python3 terminator` in that directory to see if you can reproduce the problem on master?

**Comment by jhhenriquee** [comment]
Ok.
First I uninstall terminator via pop shop, then I installed the followed package, that I saw in the install.md

sudo apt install python3-gi python3-gi-cairo python3-psutil python3-configobj gir1.2-keybinder-3.0 gir1.2-vte-2.91 gettext intltool dbus-x1

then I clone the repository, run the terminator with python3, and still got the same error.


![image](https://user-images.githubusercontent.com/71674873/132591226-2b338796-a253-4ed9-942f-c49d91c95336.png)

**Comment by jhhenriquee** [comment]
So... i tried again pulling older branches. 
And the branch issue78 worked...

Now... how can I fix this in my computer? I mean, use the terminator in that branch, without having to open the terminal and execute like that? 

Sorry if the answer seens kind obvious, I manage linux in servers since a while, but using only about one week using in my personal notebook.


![image](https://user-images.githubusercontent.com/71674873/132593263-d87f4c9a-d530-4095-8e77-9c4861cd5f87.png)

![image](https://user-images.githubusercontent.com/71674873/132593435-e0f47e1c-205d-4dc0-b959-c5bb3871d1ce.png)

**Comment by mattrose** [comment]
This bug is the bane of my existence.  https://github.com/gnome-terminator/terminator/issues/78#issuecomment-638315748 says that if you set `IBUS_NO_SNOOPER_APPS=python3` that should fix it.  I'll re-read the original and try to come up with a fix for it ... again.

**Comment by jhhenriquee** [comment]
no luck
i will read this issue78, if anything works I llet you know
thanks

![image](https://user-images.githubusercontent.com/71674873/132596092-d0d9c81e-a273-4527-9061-1f6c5813a887.png)

**Comment by jhhenriquee** [comment]
So, I'm not exactly proud of my self, but with your help pointing this branch to test, i kind solved it. Not the best way, but it works.

I keept download this version of the branch (issue78), then I ran 
```
gsettings set org.gnome.desktop.default-applications.terminal exec 'python3 /dir_with_the_old_branch/terminator'
```
Now, when I press super+t, opens the terminator that works fine. So far, so good.

Hope you can solve this bug, if you need any information that you think may help you, just ask.

**Comment by Rtoax** [comment]
why `IBUS_NO_SNOOPER_APPS=python3` not works for me?

![image](https://github.com/user-attachments/assets/30c8f4d8-1ad1-435a-979b-658596bba2bf)

**Comment by Rtoax** [comment]
Add lines in `/usr/bin/terminator` works for me in fedora41

```
## Modify to stop double character broadcast
os.environ['DBUS_SESSION_BUS_ADDRESS']=''
os.environ['IBUS_NO_SNOOPER_APPS']='python3'
## End of modify broadcast twice
```

![image](https://github.com/user-attachments/assets/fcffacda-f491-4be3-b110-e5341f4be30e)

**Comment by norman-nhan** [comment]
> Add lines in `/usr/bin/terminator` works for me in fedora41
> 
> ```
> ## Modify to stop double character broadcast
> os.environ['DBUS_SESSION_BUS_ADDRESS']=''
> os.environ['IBUS_NO_SNOOPER_APPS']='python3'
> ## End of modify broadcast twice
> ```
> 
> ![image](https://github.com/user-attachments/assets/fcffacda-f491-4be3-b110-e5341f4be30e)

I'm using ubuntu 22.04, this works for me.

**Comment by Moraisgit** [comment]
> > Add lines in `/usr/bin/terminator` works for me in fedora41
> > ```
> > ## Modify to stop double character broadcast
> > os.environ['DBUS_SESSION_BUS_ADDRESS']=''
> > os.environ['IBUS_NO_SNOOPER_APPS']='python3'
> > ## End of modify broadcast twice
> > ```
> 
> I can confirm that this fix works for me on Linux Mint 22.2 (which is based on Ubuntu 24.04) on Cinnamon desktop.

Hey, could you tell me where to insert those lines on the file please?

edit: Works for me on Ubuntu 22.04.5 LTS x86_64

<img width="1920" height="1053" alt="Image" src="https://github.com/user-attachments/assets/b0a23190-2c77-4d45-aa7a-7d8cad18b3b9" />

**Comment by BlackHoleSurvivor** [comment]
> Add lines in `/usr/bin/terminator` works for me in fedora41
> 
> ```
> ## Modify to stop double character broadcast
> os.environ['DBUS_SESSION_BUS_ADDRESS']=''
> os.environ['IBUS_NO_SNOOPER_APPS']='python3'
> ## End of modify broadcast twice
> ```
> 
> ![image](https://github.com/user-attachments/assets/fcffacda-f491-4be3-b110-e5341f4be30e)

works for me, with Ubuntu 22.04.5 & Terminator 2.1.5, X11 not Wayland
